CMakeLists.txt

```diff
@@ -979,6 +987,7 @@ set_ifndef(CSTD c99)
 zephyr_compile_options(
   $<$<COMPILE_LANGUAGE:C>:$<TARGET_PROPERTY:compiler,cstd>${CSTD}>
 )
+set(CMAKE_C_COMPILE_FEATURES ${compile_features_${CSTD}} PARENT_SCOPE)
```

Comment:
Why do we set at such a late stage?  It's already past module handing.  Thus, no module get `CMAKE_C_COMPILE_FEATURES`

Reply:
because it's the closest place to where we check if the `CSTD` setting has been specified by the user, and if not we default to `c99` if you look at the code:
https://github.com/zephyrproject-rtos/zephyr/blob/7be8d9ba4316c9fdb357e41cab7b458778192d2f/CMakeLists.txt#L983-L990

Of course we could move all this whole block further up, but as a bug fix so close to release I didn't feel like moving larger code blocks around.

I didn't expect this to be used by Zephyr modules, but in code included after `find_package(Zephyr)`.

If using compile features inside Zephyr modules is a desired use-case then I can take a closer look at this part to ensure it is safe to move up.

Reply:
There is not many modules which is based on CMake.  OpenThread is one of them, if not only one, and it doesn't use `target_compile_features()` AFAIK.  But many project use it and one I'm trying to use does.

I know it's very close to the release, so it's OK for me to go with the current version for the release.  But I appreciate if you can take a look and move it up there on the `main` branch after the release.